src/Components/Steps/Expenses/Expenses.tsx

```diff
@@ -0,0 +1,323 @@
+import { FormattedMessage, useIntl } from 'react-intl';
+import QuestionHeader from '../../QuestionComponents/QuestionHeader';
+import HelpButton from '../../HelpBubbleIcon/HelpButton';
+import QuestionQuestion from '../../QuestionComponents/QuestionQuestion';
+import QuestionDescription from '../../QuestionComponents/QuestionDescription';
+import {
+  Box,
+  Button,
+  FormControl,
+  FormControlLabel,
+  FormHelperText,
+  InputAdornment,
+  InputLabel,
+  MenuItem,
+  Radio,
+  RadioGroup,
+  Select,
+  Stack,
+  TextField,
+} from '@mui/material';
+import { Context } from '../../Wrapper/Wrapper';
+import { useContext, useEffect } from 'react';
+import { useForm, Controller, SubmitHandler, useFieldArray } from 'react-hook-form';
+import { zodResolver } from '@hookform/resolvers/zod';
+import * as z from 'zod';
+import PrevAndContinueButtons from '../../PrevAndContinueButtons/PrevAndContinueButtons';
+import { useNavigate, useParams } from 'react-router-dom';
+import { updateScreen } from '../../../Assets/updateScreen';
+import { useGoToNextStep } from '../../QuestionComponents/questionHooks';
+import { useConfig } from '../../Config/configHook';
+import { FormattedMessageType } from '../../../Types/Questions';
+import ErrorMessageWrapper from '../../ErrorMessage/ErrorMessageWrapper';
+import CloseButton from '../../CloseButton/CloseButton';
+import AddIcon from '@mui/icons-material/Add';
+import './Expenses.css';
+
+const Expenses = () => {
+  const { formData, setFormData, locale } = useContext(Context);
+  const { uuid, id } = useParams();
+  const currentStepId = Number(id);
+  const navigate = useNavigate();
+  const intl = useIntl();
+  const translatedAriaLabel = intl.formatMessage({
+    id: 'questions.hasExpenses-ariaLabel',
+    defaultMessage: 'has expenses',
+  });
+  const backNavigationFunction = () => navigate(`/${uuid}/step-${currentStepId - 1}/${formData.householdSize}`);
+  const nextStep = useGoToNextStep('hasExpenses');
+  const expenseOptions = useConfig('expense_options');
+
+  const oneOrMoreDigitsButNotAllZero = /^(?!0+$)\d+$/;
+  const expenseSourceSchema = z.object({
+    expenseSourceName: z.string().min(1),
+    expenseAmount: z.string().regex(oneOrMoreDigitsButNotAllZero),
+  });
+  const expenseSourcesSchema = z.array(expenseSourceSchema);
+  const hasExpensesSchema = z.string().regex(/^true|false$/);
```

Comment:
Setting the `hasExpenses` type to `boolean` doesn't work because MUI transforms all inputs into strings and the `zod` transform method would make it a boolean so you end up with a type error because it can't compare a string to a boolean. All of the types have to match in order for the `zod` validation to work.